src/rpc/util.cpp

```diff
@@ -117,16 +117,18 @@ CoinStatsHashType ParseHashType(const UniValue& param, const CoinStatsHashType d
 {
     if (param.isNull()) {
         return default_type;
-    } else {
-        std::string hash_type_input = param.get_str();
+    }
 
-        if (hash_type_input == "hash_serialized_2") {
-            return CoinStatsHashType::HASH_SERIALIZED;
-        } else if (hash_type_input == "none") {
-            return CoinStatsHashType::NONE;
-        } else {
-            throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("%d is not a valid hash_type", hash_type_input));
-        }
+    std::string hash_type_input = param.get_str();
+
+    if (hash_type_input == "hash_serialized_2") {
+        return CoinStatsHashType::HASH_SERIALIZED;
+    } else if (hash_type_input == "none") {
+        return CoinStatsHashType::NONE;
+    } else if (hash_type_input == "muhash") {
+        return CoinStatsHashType::MUHASH;
+    } else {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("%d is not a valid hash_type", hash_type_input));
```

Comment:
bde3febaf this could perhaps be a bit simpler

<details><summary>diff</summary><p>

```diff
diff --git a/src/rpc/blockchain.cpp b/src/rpc/blockchain.cpp
index 576dabc349..3a07e5620d 100644
--- a/src/rpc/blockchain.cpp
+++ b/src/rpc/blockchain.cpp
@@ -1079,7 +1079,7 @@ static RPCHelpMan gettxoutsetinfo()
     CCoinsStats stats;
     ::ChainstateActive().ForceFlushStateToDisk();
 
-    const CoinStatsHashType hash_type = ParseHashType(request.params[0], CoinStatsHashType::HASH_SERIALIZED);
+    const CoinStatsHashType hash_type{request.params[0].isNull() ? CoinStatsHashType::HASH_SERIALIZED : ParseHashType(request.params[0].get_str())};
 
     CCoinsView* coins_view = WITH_LOCK(cs_main, return &ChainstateActive().CoinsDB());
     NodeContext& node = EnsureNodeContext(request.context);
diff --git a/src/rpc/util.cpp b/src/rpc/util.cpp
index 30e09a9aac..d8dadf41f7 100644
--- a/src/rpc/util.cpp
+++ b/src/rpc/util.cpp
@@ -113,14 +113,8 @@ std::vector<unsigned char> ParseHexO(const UniValue& o, std::string strKey)
     return ParseHexV(find_value(o, strKey), strKey);
 }
 
-CoinStatsHashType ParseHashType(const UniValue& param, const CoinStatsHashType default_type)
+CoinStatsHashType ParseHashType(const std::string& hash_type_input)
 {
-    if (param.isNull()) {
-        return default_type;
-    }
-
-    std::string hash_type_input = param.get_str();
-
     if (hash_type_input == "hash_serialized_2") {
         return CoinStatsHashType::HASH_SERIALIZED;
     } else if (hash_type_input == "none") {
diff --git a/src/rpc/util.h b/src/rpc/util.h
index 942c243718..611d281014 100644
--- a/src/rpc/util.h
+++ b/src/rpc/util.h
@@ -77,7 +77,7 @@ extern uint256 ParseHashO(const UniValue& o, std::string strKey);
 extern std::vector<unsigned char> ParseHexV(const UniValue& v, std::string strName);
 extern std::vector<unsigned char> ParseHexO(const UniValue& o, std::string strKey);
 
-CoinStatsHashType ParseHashType(const UniValue& param, const CoinStatsHashType default_type);
+CoinStatsHashType ParseHashType(const std::string& hash_type_input);
 
 extern CAmount AmountFromValue(const UniValue& value);
 extern std::string HelpExampleCli(const std::string& methodname, const std::string& args);
```
</p></details>

Reply:
Could also drop the header file declaration and move the function into the same file as the caller, unless you expect it to have callers in other files.

Reply:
You're right, I don't foresee any other callers at the moment so I moved the function into `rpc/blockchain.cpp`.